### DISCUSSION TITLE
Support for Plugins / FunctionChoiceBehavior with AzureAssitantAgent

### DISCUSSION THREAD

**Discussion by awonglk** [microsoft/semantic-kernel#8501] [General]
Just wondering if function calls / plugins is restricted to AzureChatCompletion / OpenAIChatCompletion.
I am trying the newer AzureAssistantAgent. The "add_plugin" call seems to be ok. But if I try to use the "get_prompt_execution_settings_from_service_id()" call to set FunctionChoiceBehavior, it's coming back with:

`semantic_kernel.exceptions.kernel_exceptions.KernelServiceNotFoundError: No services found of type None
`

I haven't seen any examples (at least in Python section) where plugins is used with AzureAssistantAgent. So, I am curious if this is supported?

**Comment by moonbox3** [comment]
Hi @awonglk, you'll need to pass in the `service_id` to the `get_prompt_execution_settings_from_service_id()` method to be able to get the prompt execution settings for the AI service connector you previously defined on the kernel. But this is only applicable for Chat Completion agents, and not for Assistant agents.

Please have a look at the following Python sample that shows an (Azure)OpenAI Assistant agent using a sample plugin:

https://github.com/microsoft/semantic-kernel/blob/main/python/samples/getting_started_with_agents/step7_assistant.py

Again, note that for Assistant agents, there's no `function choice behavior` concept required. The tools (plugins) that are set on the kernel are broadcasted to OpenAI, and the tool calling is handled as part of executing the thread.
